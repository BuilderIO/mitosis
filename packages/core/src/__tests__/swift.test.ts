import { describe } from 'vitest';
import { componentToSwift } from '..';
import { runTestsForTarget } from './test-generator';

describe('swift', () => {
  const generator = componentToSwift;

  runTestsForTarget({ options: {}, only: ['SwiftPlainComponent'], target: 'swift', generator });
});
