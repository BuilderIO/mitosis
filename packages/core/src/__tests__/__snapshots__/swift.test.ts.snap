// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`swift > jsx > Only Test > SwiftPlainComponent 1`] = `
"import SwiftUI

struct MyComponent: View {

var body: some View {
  VStack(alignment: .leading, spacing: 8) {
    ForEach(0..<10, id: \\\\.self) { item in
      Text(\\"\\")
    }
    VStack(alignment: .leading, spacing: 8) {
      VStack(alignment: .leading, spacing: 8) {
        Text(\\"\\"\\"
        WHAT'S DIFFERENT ABOUT SHOPAHOLIC
        \\"\\"\\")
      }
      VStack(alignment: .leading, spacing: 8) {
        Text(\\"\\"\\"
        At Shopaholic, we're revolutionizing the online shopping experience through our commitment
        to sustainability, innovation, and customer satisfaction. Our unique approach combines
        lightning-fast delivery, eco-friendly practices, and cutting-edge technology to ensure you
        get the best products while minimizing environmental impact.
        \\"\\"\\")
      }
      VStack(alignment: .leading, spacing: 8) {
        VStack(alignment: .leading, spacing: 8) {
          VStack(alignment: .leading, spacing: 8) {

          }
          VStack(alignment: .leading, spacing: 8) {
            Text(\\"\\"\\"
            2-Day Shipping
            \\"\\"\\")
          }
          VStack(alignment: .leading, spacing: 8) {
            Text(\\"\\"\\"
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam posuere erat a ante
            vestibulum, in volutpat ligula elementum.
            \\"\\"\\")
          }
        }
        VStack(alignment: .leading, spacing: 8) {
          Image(\\"placeholder\\").resizable().aspectRatio(contentMode: .fit)
          VStack(alignment: .leading, spacing: 8) {
            Text(\\"\\"\\"
            Carbon Neutral
            \\"\\"\\")
          }
          VStack(alignment: .leading, spacing: 8) {
            Text(\\"\\"\\"
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam posuere erat a ante
            vestibulum, in volutpat ligula elementum.
            \\"\\"\\")
          }
        }
        VStack(alignment: .leading, spacing: 8) {
          Image(\\"placeholder\\").resizable().aspectRatio(contentMode: .fit)
          VStack(alignment: .leading, spacing: 8) {
            Text(\\"\\"\\"
            Advanced Dye Tech
            \\"\\"\\")
          }
          VStack(alignment: .leading, spacing: 8) {
            Text(\\"\\"\\"
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam posuere erat a ante
            vestibulum, in volutpat ligula elementum.
            \\"\\"\\")
          }
        }
      }
    }
  }

  }

  }

  #if DEBUG
  struct MyComponent_Previews: PreviewProvider {
    static var previews: some View {
      MyComponent(

      )
    }
  }
  #endif"
`;
